### PR TITLE
Fix #18628 and other smallish WS changes

### LINKED ISSUE #18628: investigate handling of UserHandlerErredOut in ws.FrameOutHandler
there is currently no test that exercises it

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #18628 and #18759.

/cc @sirthias

### CODE REVIEW

**Comment by jrudolph** [comment]
Rebased and added fix fo #18759.

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/Websocket.scala [hunk 1]
```diff
@@ -68,7 +68,7 @@ private[http] object Websocket {
       var inMessage = false
       def onPush(elem: MessagePart, ctx: Context[MessagePart]): SyncDirective = elem match {
         case PeerClosed(code, reason) ⇒
-          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new ProtocolException(s"Peer closed connection with code $code"))
+          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new PeerClosedConnectionException(code.get, reason))
           else if (inMessage) ctx.fail(new ProtocolException(s"Truncated message, peer closed connection in the middle of message."))
           else ctx.finish()
         case ActivelyCloseWithCode(code, reason) ⇒
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/Websocket.scala]: Can't comment on line `.splitWhen(_.isMessageEnd) // FIXME using splitAfter from #16885 would simplify protocol a lot` https://github.com/akka/akka/pull/18750/files#diff-32111086a8e0b3bf2b812bda44d7c0a1R110 but that should be solved now https://github.com/akka/akka/issues/16885

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameOutHandler.scala [hunk 2]
```diff
@@ -43,7 +43,7 @@ private[http] class FrameOutHandler(serverSide: Boolean, _closeTimeout: FiniteDu
       case UserHandlerCompleted ⇒
         become(new WaitingForPeerCloseFrame())
         ctx.push(FrameEvent.closeFrame(Protocol.CloseCodes.Regular))
-      case UserHandlerErredOut(ex) ⇒
+      case UserHandlerErredOut(_) ⇒
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameOutHandler.scala]: Added a commit that logs an error here.

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/Websocket.scala [hunk 3]
```diff
@@ -68,7 +71,7 @@ private[http] object Websocket {
       var inMessage = false
       def onPush(elem: MessagePart, ctx: Context[MessagePart]): SyncDirective = elem match {
         case PeerClosed(code, reason) ⇒
-          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new ProtocolException(s"Peer closed connection with code $code"))
+          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new PeerClosedConnectionException(code.get, reason))
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/Websocket.scala]: :+1:

**Comment by ktoso** [comment]
LGTM, will merge after validation

**Comment by jrudolph** [comment]
Is jenkins not putting any PASSED notifications out any more?

**Comment by ktoso** [comment]
I updated plugins yesterday, something must have changed. I added some settings now around this, should be reporting again.

**Comment by ktoso** [comment]
Thanks!

**Pull request merged**